### PR TITLE
SourceSetData fixes

### PR DESCRIPTION
**Pull request by kamildoleglo**
Remove SourceSetDataCache, rename PassConfiguration to DokkaSourceSet and use it instead of SourceSetData

### CODE REVIEW

File: core/src/main/kotlin/model/Documentable.kt [hunk 1]
```diff
@@ -1,20 +1,22 @@
 package org.jetbrains.dokka.model
 
 import com.intellij.psi.PsiNamedElement
+import org.jetbrains.dokka.DokkaConfiguration
```
Comment (sellophane) [inline core/src/main/kotlin/model/Documentable.kt:4]: This one is needed?

File: plugins/base/src/main/kotlin/renderers/html/HtmlRenderer.kt [hunk 2]
```diff
@@ -1,13 +1,15 @@
 package org.jetbrains.dokka.base.renderers.html
 
-import kotlinx.coroutines.*
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
```
Comment (sellophane) [inline plugins/base/src/main/kotlin/renderers/html/HtmlRenderer.kt:5]: Are these used?

File: plugins/base/src/main/kotlin/translators/psi/DefaultPsiToDocumentableTranslator.kt [hunk 3]
```diff
@@ -1,14 +1,13 @@
 package org.jetbrains.dokka.base.translators.psi
 
-import com.intellij.icons.AllIcons.Nodes.Static
-import com.intellij.lang.jvm.annotation.JvmAnnotationAttribute
 import com.intellij.lang.jvm.JvmModifier
+import com.intellij.lang.jvm.annotation.JvmAnnotationAttribute
```
Comment (sellophane) [inline plugins/base/src/main/kotlin/translators/psi/DefaultPsiToDocumentableTranslator.kt:4]: Do we need it?

File: core/src/main/kotlin/model/Documentable.kt (hunk 1 above)
Comment (kamildoleglo) [reply core/src/main/kotlin/model/Documentable.kt:4]: Fixed

File: plugins/base/src/main/kotlin/renderers/html/HtmlRenderer.kt (hunk 2 above)
Comment (kamildoleglo) [reply plugins/base/src/main/kotlin/renderers/html/HtmlRenderer.kt:5]: Yes, the HtmlRenderer is asynchronous

File: plugins/base/src/main/kotlin/translators/psi/DefaultPsiToDocumentableTranslator.kt (hunk 3 above)
Comment (kamildoleglo) [reply plugins/base/src/main/kotlin/translators/psi/DefaultPsiToDocumentableTranslator.kt:4]: Yes, for extracting annotations' attributes

File: core/src/main/kotlin/DokkaGenerator.kt [hunk 4]
```diff
@@ -152,7 +145,7 @@ class DokkaGenerator(
 
     private fun createEnvironmentAndFacade(
         configuration: DokkaConfiguration,
-        pass: DokkaConfiguration.PassConfiguration
+        pass: DokkaSourceSet
```
Comment (sellmair) [inline core/src/main/kotlin/DokkaGenerator.kt:148]: Should we stick with this concept of naming something "pass" 🤔

File: plugins/base/src/main/kotlin/renderers/html/NavigationPage.kt [hunk 5]
```diff
@@ -41,7 +41,7 @@ class NavigationPage(val root: NavigationNode) : RendererSpecificPage {
 class NavigationNode(
     val name: String,
     val dri: DRI,
-    val platforms: List<SourceSetData>,
+    val platforms: List<DokkaSourceSet>,
```
Comment (sellmair) [inline plugins/base/src/main/kotlin/renderers/html/NavigationPage.kt:44]: Maybe rename this to "sourceSets"?

Comment (kamildoleglo) [reply plugins/base/src/main/kotlin/renderers/html/NavigationPage.kt]: Fixed

File: core/src/main/kotlin/model/documentableUtils.kt [hunk 6]
```diff
@@ -1,9 +1,11 @@
 package org.jetbrains.dokka.model
 
-fun <T> SourceSetDependent<T>.filtered(platformDataList: Set<SourceSetData>) = filter { it.key in platformDataList }
-fun SourceSetData?.filtered(platformDataList: Set<SourceSetData>) = takeIf { this in platformDataList }
+import org.jetbrains.dokka.DokkaConfiguration.DokkaSourceSet
 
-fun DTypeParameter.filter(filteredData: Set<SourceSetData>) =
+fun <T> SourceSetDependent<T>.filtered(platformDataList: Set<DokkaSourceSet>) = filter { it.key in platformDataList }
```
[the remark this answers is not in the document]
Comment (kamildoleglo) [reply core/src/main/kotlin/model/documentableUtils.kt]: Changed

File: core/src/main/kotlin/DokkaGenerator.kt (hunk 4 above)
Comment (kamildoleglo) [reply core/src/main/kotlin/DokkaGenerator.kt]: Changed

**Pull request merged**